src/Dashboard/Filter.php

```diff
@@ -58,235 +60,39 @@ class Filter extends \CommonDBChild
 
     /**
      * Return all available filters
-     * Plugins can hooks on this functions to add their own filters
      *
      * @return array of filters
      */
     public static function getAll(): array
     {
+        global $PLUGIN_HOOKS;
+        $more_filters = [];
+
         $filters = [
-            'dates'        => __("Creation date"),
-            'dates_mod'    => __("Last update"),
-            'itilcategory' => ITILCategory::getTypeName(Session::getPluralNumber()),
-            'requesttype'  => RequestType::getTypeName(Session::getPluralNumber()),
-            'location'     => Location::getTypeName(Session::getPluralNumber()),
-            'manufacturer' => Manufacturer::getTypeName(Session::getPluralNumber()),
-            'state'        => State::getTypeName(Session::getPluralNumber()),
-            'tickettype'   => _n("Ticket type", "Ticket types", Session::getPluralNumber()),
-            'group_tech'   => __("Technician group"),
-            'user_tech'    => __("Technician"),
+            DatesFilter::class,
+            DatesModFilter::class,
+            ItilCategoryFilter::class,
+            LocationFilter::class,
+            ManufacturerFilter::class,
+            RequestTypeFilter::class,
+            StateFilter::class,
+            TicketTypeFilter::class,
+            GroupTechFilter::class,
+            UserTechFilter::class,
         ];
 
-        $more_filters = Plugin::doHookFunction(Hooks::DASHBOARD_FILTERS);
-        if (is_array($more_filters)) {
-            $filters = array_merge($filters, $more_filters);
+        if (isset($PLUGIN_HOOKS[Hooks::DASHBOARD_FILTERS])) {
+            foreach ($PLUGIN_HOOKS[Hooks::DASHBOARD_FILTERS] as $hook_filters) {
+                foreach ($hook_filters as $filter) {
+                    $more_filters[] = $filter;
+                }
+            }
         }
+        $filters = array_merge($filters, $more_filters);
 
         return $filters;
     }
 
-    /**
-     * Get HTML for a dates range filter
-     *
-     * @param string|array $values init the input with these values, will be a string if empty values
-     * @param string $fieldname how is named the current date field
-     *                         (used to specify creation date or last update)
-     *
-     * @return string
-     */
-    public static function dates($values = "", string $fieldname = 'dates'): string
```

Comment:
Public methods should not be removed on a bugfixes version. At least, it should trigger a deprecation notice and maybe return a mocked value.